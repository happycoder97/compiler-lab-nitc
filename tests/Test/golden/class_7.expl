class A         
{
    int foo;
    int setFoo(int f) {
        write("SET");
        write(f);
        self.foo = f;
    }

    int printFoo() {
        write(self.foo);
    }
}


class B extends A
{
    int setFoo(int f) {
        write("SET B");
        write(f);
        self.foo = f;
    }

    int printFooB() {
        write("PRINTB");
        write(self.foo);
    }
}

A a;
B b;

int main() {
    initialize();
    a = new(B);
    a.setFoo(123);
    a.printFoo();

    b = new(B);
    b.setFoo(123);
    b.printFoo();
    b.printFooB();
    write("DONE");
}
