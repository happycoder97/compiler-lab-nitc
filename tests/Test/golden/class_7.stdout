SET B
123
123
SET B
123
123
PRINTB
123
DONE
